Makefile

```diff
@@ -762,24 +769,43 @@ runtime/ld.conf: $(ROOTDIR)/Makefile.config
 # see http://pubs.opengroup.org/onlinepubs/9699919799/utilities/sort.html:
 # "using sort to process pathnames, it is recommended that LC_ALL .. set to C"
 
-# To speed up builds, we avoid changing "primitives" when files
-# containing primitives change but the primitives table does not
-runtime/primitives: \
-  $(shell runtime/gen_primitives.sh > runtime/primitives.new; \
-                    cmp -s runtime/primitives runtime/primitives.new || \
-                    echo runtime/primitives.new)
-	cp $^ $@
+# To speed up builds, we avoid changing "primitives" and
+# "primitive-arguments" when files containing primitives change but
+# the primitives table does not
+runtime/primitives-arguments.new: \
+  $(runtime_BUILT_HEADERS) \
+  $(runtime_PRIMITIVES) \
```

Comment:
You're moving code from a script into the toplevel Makefile.  I'm not sure this is a good idea.

Reply:
It should be ok: we switch from one trick to use a script to conditionally add a prerequisite or not, which felt like advanced magic, to resetting the modified time of a file.
It does break the Dune build, though.

Reply:
The Dune build could be a concern.  But my more general concern is that too much scripting within a Makefile renders the Makefile unreadable; at some point, having separate scripts that do one thing but do it well becomes preferable.

Reply:
I agree, that's why I've externalised the sed scripts.
However if we chose to keep using the preprocessor, I think it'll be difficult to give the `$(CPP)` content to the script due to usual concerns of spaces and word-splitting.